doc/source/development/rfc/rfc86_column_oriented_api.rst

```diff
@@ -0,0 +1,692 @@
+closing). Furthermore, unless otherwise specified by a particular driver
+implementation, only one ArrowArrayStream can be active at a time on
+a given layer (that is the last active one must be explicitly released before
+a next one is asked). Changing filter state, ignored columns, modifying the schema
+or using ResetReading()/GetNextFeature() while using a ArrowArrayStream is
+strongly discouraged and may lead to unexpected results. As a rule of thumb,
+no OGRLayer methods that affect the state of a layer should be called on a
+layer, while an ArrowArrayStream on it is active.
+
+A potential usage can be:
+
+.. code-block:: cpp
+
+    struct ArrowArrayStream stream;
+    if( !poLayer->GetArrowStream(&stream, nullptr))
+    {
+        fprintf(stderr, "GetArrowStream() failed\n");
+        exit(1);
+    }
+    struct ArrowSchema schema;
+    if( stream.get_schema(&stream, &schema) == 0 )
+    {
+        // Do something useful
+        schema.release(schema);
+    }
+    while( true )
+    {
+        struct ArrowArray array;
+        // Look for an error (get_next() returning a non-zero code), or
+        // end of iteration (array.release == nullptr)
+        //
+        if( stream.get_next(&stream, &array) != 0 ||
+            array.release == nullptr )
+        {
+            break;
+        }
+        // Do something useful
+        array.release(&array);
+    }
+    stream.release(&stream);
+
+The papszOptions that may be provided is a NULL terminated list of key=value
+strings, that may be driver specific.
+
+OGRLayer has a base implementation of GetArrowStream() that is such:
+
+- The get_schema() callback returns a schema whose top-level object returned is
+  of type Struct, and whose children consist in the FID column, all OGR attribute
+  fields and geometry fields to Arrow fields.
+  The FID column may be omitted by providing the INCLUDE_FID=NO option.
+
+  When get_schema() returns 0, and the schema is no longer needed, it must
+  be released with the following procedure, to take into account that it might
+  have been released by other code, as documented in the Arrow C data
+  interface:
+
+  .. code-block:: c
+
+          if( out_schema->release )
+              out_schema->release(out_schema)
+
+
+- The get_next() callback retrieve the next record batch over the layer.
+
+  out_array is a pointer to a ArrowArray structure, that can be in a uninitialized
+  state (the method will ignore any initial content).
+
+  The default implementation uses GetNextFeature() internally to retrieve batches
+  of up to 65,536 features (configurable with the MAX_FEATURES_IN_BATCH=num option).
+  The starting address of buffers allocated by the
+  default implementation is aligned on 64-byte boundaries.
+
+  The default implementation outputs geometries as WKB in a binary field,
+  whose corresponding entry in the schema is marked with the metadata item
+  ``ARROW:extension:name`` set to ``WKB``. Specialized implementations may output
```

Comment:
The general recommendation at https://arrow.apache.org/docs/dev/format/Columnar.html#extension-types is to "namespace" the name value. I don't know if "WKB" is an abbreviation that is also used in other context, but so that is something to consider here.

One option it to use the "geoarrow" namespace (eg "geoarrow.wkb"); we are also considering to keep this as an option in the geo-arrow-spec (current PR to list those names is at https://github.com/geopandas/geo-arrow-spec/pull/22.

On the one hand that keeps it consistent with the other encodings ("geoarrow.point", "geoarrow.linestring", ..), but on the other hand this can also be confusing if we are using "geoarrow" for both the nested-list-array based encodings as for a binary one like WKB (I know that @kylebarron has mentioned this confusion before).

Using something like "gdal.wkb" is also an option, but on the other hand the representation itself (WKB) is not really GDAL specific, so it might be nice to have a more general name.

Reply:
(this is actually also something that is relevant for the existing Arrow driver, so it might be worth to have this conversation in a separate issue)

Reply:
It might be "ogc.wkb" but some parties think that the axis order in WKB should depend on CRS  https://bugs.mysql.com/bug.php?id=88869. So perhaps gdal.wkb is better, or ogc.wkb-subversion-as-in-the-gpkg-standard

> [K11] The axis order in WKB is always (x,y{,z}{,m}) where x is easting or longitude, y is northing
> or latitude, z is optional elevation and m is optional measure.

Reply:
The coordinate order and the WKB encoding are orthogonal topics (GDAL has its own mechanics to possibly define an for the override the CRS axis order through the DataAxisToSRSAxisMapping concept: https://gdal.org/tutorials/osr_api_tut.html#crs-and-axis-order)
==> "ogc.wkb" or "iso.wkb" could be good candidates

Reply:
changed to "ogc.wkb"